torchx/specs/api.py

```diff
@@ -219,11 +219,13 @@ class RetryPolicy(str, Enum):
                 application to deal with failed replica departures and
                 replacement replica admittance.
     2. APPLICATION: Restarts the entire application.
-
+    3. QUORUM: Restarts the replicas for a role as long as the quorum is not
+               violated. (EXPERIMENTAL)
     """
 
     REPLICA = "REPLICA"
     APPLICATION = "APPLICATION"
+    QUORUM = "QUORUM"
```

Comment:
will the quorum restart policy be checked in? I'm only seeing the specs changes here.

Reply:
Its a meta only change to some schedulers there is nothing that has to be handled for existing schedulers

Reply:
For existing open source schedulers (i.e. Ray/Volcano) that support some form of elasticity we use `num_replicas` and `min_replicas` fields below to detect elastic behavior

Could we use that as a trigger for this behavior instead?

I'm not totally opposed to adding in a new restart policy but it's unclear what the defined semantics for this is. Currently RetryPolicy defines service vs gang scheduling -- it's not clear that quorum is orthogonal behavior to that.

Reply:
Updated it to APPLICATION_HOT_SPARES just to differentiate between this restart policy and elasticity in general for now. We can see how things evolve on the MAST side and what version of elaticity of replacement we end up with

Reply:
@kiukchung I think this is generic enough and now well defined enough to be included in the core APIs. Nothing else supports it currently but if anyone uses it it'll just throw an error.

https://github.com/pytorch/torchx/blob/main/torchx/schedulers/kubernetes_scheduler.py#L396

We could throw a better error rather than just KeyError